### ISSUE TITLE
How to run a double input onnx model

### ISSUE THREAD

**Issue by muyi6** [microsoft/onnxruntime#11453]
**Describe the bug**
When using 'Conv' operator with Double Tensor type the following error occurs:
sess.initialize_session(providers, provider_options) onnxruntime.capi.onnxruntime_pybind11_state.NotImplemented: [ONNXRuntimeError] : 9 : NOT_IMPLEMENTED : Could not find an implementation for the node Conv(11) node with name ''

**System information**
Python 3.8.5 on Ubuntu-18.04.1
onnx: 1.9.0
onnxruntime: 1.8.0

**To Reproduce**
data_input=helper.make_tensor_value_info('data_input',TensorProto.DOUBLE, (26,10,13,21)) 
output=helper.make_tensor_value_info('output',TensorProto.DOUBLE,(26,43,13,21)) 
op_type ='Conv'
node=onnx.helper.make_node(
op_type,
inputs=['data_input','W'],
outputs=['output'],
)
W_const=np.random.randn(43, 10, 1, 1).astype(np.float64)
W_initializer=onnx.helper.make_tensor("W", onnx.TensorProto.DOUBLE,W_const.shape, W_const.flatten().tolist()) 
graph =helper.make_graph([node], 'demo_conv_graph', [data_input], [output]) 
model=helper.make_model(graph, producer_name='demo_model') 
model.graph.initializer.append(W_initializer) 
onnx.save(model, 'test_conv_default.onnx')

ort_session = onnxruntime.InferenceSession('test_conv_default.onnx')
data = np.random.randn(26, 10, 13, 21).astype(np.float64)
input_data = {'data_input': data}
outputs = srt_session.run(None, input_data)

**Expected behavior**
ONNX should support tensor(double) type for Conv, but got error

**Screenshots**

**Additional context**
What should i do if i want to run a double onnx model

**Comment by hariharans29** [comment]
Right. Currently, we do not support ALL types for ALL operators in the ONNX standard to keep binary size in check. For Conv, it seems like we only support float - https://github.com/microsoft/onnxruntime/blob/2a90922f01e2fa9861dda0c7a769cfed09658167/onnxruntime/core/providers/cpu/nn/conv.cc#L306. You can add support for double similar to the line I referenced and build from source. 

If there is enough interest from the community, it may be added in a future release.

**Comment by muyi6** [comment]
@hariharans29 ok. got it, thanks for you quick response. For other operators like pool, bn etc，should i use the same way like 'build from source' if want to support double

**Comment by skottmckay** [comment]
Why do you need to use double instead of float? 

It costs:
  - increased processing time handling 8 byte double vs 4 byte float values
  - model uses more memory 
    - input data and weights would all be 2x larger
  - increased ORT binary size

The potential benefit is greater accuracy. 

Given we don't currently support double for Conv that would indicate current production scenarios have not required the usage of double for Conv.

So instead of building from source and manually changing code, what's the reason you can't simply use float?
